content/en/blog/_posts/2021-11-02-csi-migration-status.md

```diff
@@ -0,0 +1,130 @@
+
+If you are a user of public cloud provider managed Kubernetes service (such as AKS, EKS, or GKE) there is nothing required to be done from your end. Your cloud provider should manage the control plane configuration for you and make a seamless migration possible.
+
+If you use a distribution of Kubernetes, check its official documentation for information about support for ths feature. For the CSI Migration feature graduation to GA, SIG Storage and SIG Cluster Lifecycle are collaborating towards making the migration mechanisms available in tooling (such as kubeadm) as soon as they're available in Kubernetes itself.
+
+## What is the timeline/status?
+
+The current and targeted releases for each individual driver is shown in the table below:
+
+| Driver           | Alpha | Beta (in-tree deprecated) | Beta (on-by-default) | GA            | Target "in-tree plugin" removal |
+| ---------------- | ----- | ------------------------- | -------------------- | ------------- | ------------------------------- |
+| AWS EBS          | 1.14  | 1.17                      | 1.23                 | 1.24 (Target) | 1.26 (Target)                   |
+| GCE PD           | 1.14  | 1.17                      | 1.23                 | 1.24 (Target) | 1.26 (Target)                   |
+| OpenStack Cinder | 1.14  | 1.18                      | 1.21                 | 1.24 (Target) | 1.26 (Target)                   |
+| Azure Disk       | 1.15  | 1.19                      | 1.23                 | 1.24 (Target) | 1.26 (Target)                   |
+| Azure File       | 1.15  | 1.21                      | 1.24 (Target)        | 1.25 (Target) | 1.27 (Target)                   |
+| vSphere          | 1.18  | 1.19                      | 1.24 (Target)        | 1.25 (Target) | 1.27 (Target)                   |
+| Ceph RBD         | 1.23  |
+| Portworx         | 1.23  |
+
+The following drivers will not have CSI migration support and are deprecated and will be removed from the core Kubernetes.
+
+| Driver    | Deprecated | Code Removal  |
+| --------- | ---------- | ------------- |
+| ScaleIO   | 1.16       | 1.22          |
+| Flocker   | 1.22       | 1.25 (Target) |
+| Quobyte   | 1.22       | 1.25 (Target) |
+| StorageOS | 1.22       | 1.25 (Target) |
+
+## What's next?
+
+With more CSI drivers graduating to GA, we will soon mark the overall CSI Migration feature as GA. We are expecting cloud provider in-tree storage plugins code removal to happen by Kubernetes v1.26 and v1.27.
+
+## What should I do as a user?
+
+Note that all new features for the Kubernetes storage system (like volume snapshotting) will only be added to the CSI interface. Therefore, if you are starting up a new cluster, creating stateful applications for the first time, or require these new features we recommend using CSI drivers natively (instead of the in-tree volume plugin API). Follow the [updated user guides for CSI drivers](https://kubernetes-csi.github.io/docs/drivers.html) and use the new CSI APIs.
```

Comment:
nit:
```suggestion
Note that all new features for the Kubernetes storage system (such as volume snapshotting) will only be added to the CSI interface. Therefore, if you are starting up a new cluster, creating stateful applications for the first time, or require these new features we recommend using CSI drivers natively (instead of the in-tree volume plugin API). Follow the updated [user guides for CSI drivers](https://kubernetes-csi.github.io/docs/drivers.html) and use the new CSI APIs.
```